# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmValidate.cpp
# Commit: 02ce4be46831
# Full Hash: 02ce4be46831d77239a749bd74b79a3e1447df0a
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-18 09:24:52
# Regressor Bug: 1561521
# File Overlap Count: 1
# Description:
#   Bug 1561521 - Refine ref.null decoding to decode heaptype instead of reftype. r=lth
#   
#   ref.null takes a heaptype immediate. This is backwards compatible for
#   reference-types, but is a breaking change for the GC proposal. This commit
#   implements decoding for this refinement.
# ==============================================================================

diff -r 00d2290648a7 -r 02ce4be46831 js/src/wasm/WasmValidate.cpp
--- a/js/src/wasm/WasmValidate.cpp	Tue Aug 18 00:40:20 2020 +0000
+++ b/js/src/wasm/WasmValidate.cpp	Tue Aug 18 00:37:28 2020 +0000
@@ -2285,7 +2285,7 @@
     case uint16_t(Op::RefNull): {
       MOZ_ASSERT_IF(env->isStructType(expected), env->gcTypesEnabled());
       RefType initType;
-      if (!d.readRefType(env->types, env->gcTypesEnabled(), &initType)) {
+      if (!d.readHeapType(env->types, env->gcTypesEnabled(), true, &initType)) {
         return false;
       }
       if (!expected.isReference() ||